### DISCUSSION TITLE
Livewire Converts Searched Int to Float

### DISCUSSION THREAD

**Discussion by doubleprincez** [livewire/livewire#4286] [1. Help]
I'm trying to use livewire to fetch a product using its ID or UUID. I tried dumping the search request and I'm getting back the correct query when using Laravel request() function but livewire returns a float value instead.

## Test
I have this query in my bookshop to fetch using the ID or UUID provided
```
http://127.0.0.1:8000/bookshop?search=184742960601636827437&type=book&direction=asc&order=id
```
When I dump this using the following code in the  livewire component

```
dd($this->search, request()->get('search'), filter_var($this->search, FILTER_SANITIZE_STRING));
```
I got the following Results
 
![Fullscreen capture 11262021 31548 PM bmp](https://user-images.githubusercontent.com/37469720/143595662-0c3c601f-05e9-4c0d-9323-05e4f349b5ef.jpg)

## Conclusion
The above query will be converted to float by the livewire engine while if you use request()->get('search') will return the desired result.
Is there a way to ensure any Integer values provided as query parameters gets treated as string to ensure the desired result?

**Comment by jonneroelofs** [comment]
Not sure if this helps, but I think you probably don't want to use the request() helper in combination with Livewire, it behaves differently than in plain Laravel when you are working with controllers. 

Try this instead:
https://laravel-livewire.com/docs/2.x/query-string
